packages/starknet-snap/src/ui/components/ExecuteTxnUI.tsx

```diff
@@ -0,0 +1,131 @@
+import type { SnapComponent } from '@metamask/snaps-sdk/jsx';
+import {
+  Box,
+  Container,
+  Section,
+  Text,
+  Icon,
+  Divider,
+} from '@metamask/snaps-sdk/jsx';
+
+import type { FeeToken } from '../../types/snapApi';
+import type { FormattedCallData } from '../../types/snapState';
+import { DEFAULT_DECIMAL_PLACES } from '../../utils/constants';
+import { AddressUI, JsonDataUI } from '../fragments';
+import { Amount } from '../fragments/Amount';
+import { FeeTokenSelector } from '../fragments/FeeTokenSelector';
+import { accumulateTotals } from '../utils';
+
+export type ExecuteTxnUIProps = {
+  type: string;
+  signer: string;
+  chainId: string;
+  maxFee: string;
+  calls: FormattedCallData[];
+  selectedFeeToken: string;
+  includeDeploy: boolean;
+};
+
+/**
+ * A component for executing transactions, providing details and options to configure the transaction.
+ *
+ * @param props - The component props.
+ * @param props.signer - The signer for the transaction.
+ * @param props.chainId - The ID of the chain for the transaction.
+ * @param props.maxFee - The maximum fee allowed for the transaction.
+ * @param props.calls - The calls involved in the transaction.
+ * @param props.selectedFeeToken - The token used for fees.
+ * @param props.includeDeploy - Whether to include account deployment in the transaction.
+ * //param props.errors : TODO : add this param
+ * @returns The ExecuteTxnUI component.
+ */
+export const ExecuteTxnUI: SnapComponent<ExecuteTxnUIProps> = ({
+  signer,
+  chainId,
+  maxFee,
+  calls,
+  selectedFeeToken,
+  includeDeploy,
+  // errors, // TODO: include this later
+}) => {
+  // Calculate the totals using the helper
+  const tokenTotals = accumulateTotals(calls, maxFee, selectedFeeToken);
+
+  return (
+    <Container>
+      <Box>
+        <Section>
+          <AddressUI label="Signer" address={signer} chainId={chainId} />
```

Comment:
missing network, and reuse signer UI (not sure if the props is correct)
```suggestion
          <NetworkUI networkName={networkName} />
          <SignerUI address={signer} chainId={chainId} />
```